src/coreclr/jit/morph.cpp

```diff
@@ -1865,16 +1832,16 @@ TempInfo Compiler::fgMakeTemp(GenTree* rhs, CORINFO_CLASS_HANDLE structType /*=
 //    A fresh GT_LCL_VAR node referencing the temp which has not been used
 //
 // Notes:
-//    Caller must ensure that if the node is an unaliased local, the second use this
-//    creates will be evaluated before the local can be reassigned.
-//
-//    Can be safely called in morph preorder, before GTF_GLOB_REF is reliable.
+//    This function will clone invariant nodes and locals, so this function
+//    should only be used in situations where no interference between the
+//    original use and new use is possible. Otherwise, fgInsertCommaFormTemp
+//    should be used directly.
 //
 GenTree* Compiler::fgMakeMultiUse(GenTree** pOp, CORINFO_CLASS_HANDLE structType /*= nullptr*/)
 {
     GenTree* const tree = *pOp;
 
-    if (fgIsSafeToClone(tree))
+    if (tree->IsInvariant() || tree->OperIsLocal())
```

Comment:
Does a simple `IsLeaf` work here?

Reply:
`IsInvariant()` also checks for addresses of locals which may not be a leaf node. But yes, we could probably clone any leaf or even more trees with the right profitability checks.

Reply:
I saw no extra diffs from also cloning all leaves, so will leave this as is (actually, will change the use of `gtClone` below to `gtCloneExpr`).